### ISSUE TITLE
Race condition using Olric v0.5.0

### ISSUE THREAD

**Issue by darkweak** [buraksezer/olric#193]
Hello, there is a race condition in the redcon library used by olric v0.5.0.

Sample
```
	olricInstance = config.New("local")
	db, err := olric.New(olricInstance)
	if err != nil {
		return nil, err
	}

	go func() {
		_ = db.Start()
	}()
```

Output
```
WARNING: DATA RACE
Read at 0x00c0000360a8 by goroutine 19:
  github.com/tidwall/redcon.(*Writer).Flush()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:651 +0x3e
  github.com/tidwall/redcon.(*conn).Close()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:459 +0x47
  github.com/tidwall/redcon.serve.func1.1()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:338 +0x119
  github.com/tidwall/redcon.serve.func1()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:341 +0x56
  runtime.deferreturn()
      /usr/local/go/src/runtime/panic.go:476 +0x32
  github.com/tidwall/redcon.(*Server).Serve()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:311 +0x155
  github.com/buraksezer/olric/internal/server.(*Server).ListenAndServe()
      /Users/sylvain/go/pkg/mod/github.com/buraksezer/olric@v0.5.0/internal/server/server.go:189 +0x784
  github.com/buraksezer/olric.(*Olric).Start.func1()
      /Users/sylvain/go/pkg/mod/github.com/buraksezer/olric@v0.5.0/olric.go:324 +0x47
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/sylvain/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x86

Previous write at 0x00c0000360a8 by goroutine 37:
  github.com/tidwall/redcon.(*Writer).Flush()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:654 +0xbe
  github.com/tidwall/redcon.handle.func2()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:438 +0x237
  github.com/tidwall/redcon.handle()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:442 +0xb8
  github.com/tidwall/redcon.serve.func2()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:378 +0x47

Goroutine 19 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/sylvain/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0x12e
  github.com/buraksezer/olric.(*Olric).Start()
      /Users/sylvain/go/pkg/mod/github.com/buraksezer/olric@v0.5.0/olric.go:323 +0x2a4
  command-line-arguments.EmbeddedOlricConnectionFactory.func3()
      /Users/sylvain/go/src/github.com/darkweak/souin/cache/providers/embeddedOlricProvider.go:86 +0x39

Goroutine 37 (running) created at:
  github.com/tidwall/redcon.serve()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:378 +0x89b
  github.com/tidwall/redcon.(*Server).Serve()
      /Users/sylvain/go/pkg/mod/github.com/tidwall/redcon@v1.6.0/redcon.go:311 +0x155
  github.com/buraksezer/olric/internal/server.(*Server).ListenAndServe()
      /Users/sylvain/go/pkg/mod/github.com/buraksezer/olric@v0.5.0/internal/server/server.go:189 +0x784
  github.com/buraksezer/olric.(*Olric).Start.func1()
      /Users/sylvain/go/pkg/mod/github.com/buraksezer/olric@v0.5.0/olric.go:324 +0x47
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/sylvain/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x86
```

**Comment by buraksezer** [comment]
I'm aware of this problem but it seems that the race condition is caused by [tidwall/redcon](https://github.com/tidwall/redcon). We can report this bug to him.

**Comment by derekperkins** [comment]
This was reported here https://github.com/tidwall/redcon/issues/61, but they were unable to reproduce it

**Comment by buraksezer** [comment]
I provided a code sample to reproduce the problem. cc: @derekperkins and @darkweak

**Comment by buraksezer** [comment]
This problem has been fixed by Olric v0.5.2.
